launcher/src/main/go/build.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+
+SOURCE=${BASH_SOURCE[0]}
+while [ -L "$SOURCE" ]; do
+  DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )
+  SOURCE=$(readlink "$SOURCE")
+  [[ $SOURCE != /* ]] && SOURCE=$DIR/$SOURCE
+done
+
+DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )
+
+cd "${DIR}" || exit
+
+OUTPUT_DIR="../resources/binaries/"
+EXECUTABLE_NAME="launcher2"
+
+go get launcher2
+
+echo "Output directory: ${DIR}/${OUTPUT_DIR}"
+
+build_for_os_and_arch() {
+    os="${1}"
+    arch="${2}"
+    path="${1}-${2}"
+
+  echo "Building for ${os}/${arch} into ${path}/${EXECUTABLE_NAME}"
+  CGO_ENABLED=0 GOOS="${os}" GOARCH="${arch}" go build -ldflags="-s -w -extldflags=-static -X launcher2/args.Version=$(git describe HEAD)" -o "${DIR}/${OUTPUT_DIR}${path}/${EXECUTABLE_NAME}" .
```

Comment:
`git describe` is currently failing in the CI, since we don't fetch tags on checkout. This won't be an issue during release, but do we want to also add `fetch-depth: 0` in the regular CI workflow? Seeing fatal errors in logs might be confusing.

Reply:
How big are the executables? We can consider also compressing them with `upx` here.

Reply:
Upx added